.github/actions/assert-build/action.yml

```diff
@@ -0,0 +1,30 @@
+name: Build and Assert Assets Exists
+description: Build the package and assert that file contents exist as we expect
+runs:
+  using: "composite"
+  steps:
+  - name: Build and Assert Output
+    shell: bash
+    run: |-
+      echo '
+        target: ./addon/dist
+        setup:
+          run: pnpm build
+          cwd: ./addon/
+        expect: |
```

Comment:
this mostly ensures:
 - build succeeds
 - sourcemaps are emitted
 
 when TS is added, we also test:
  - if declarations are added
  - if declaration maps are added